source/index.js

```diff
@@ -35,17 +38,30 @@ export default class Chronicle {
     };
 
     this.color = new Color();
+    this.typeOptions = [];
 
     // create direct convenience methods for logging
     for (const type of Object.keys(logs)) {
-      this.setColorForType(type, logs[type]);
+      this.defineType(type, logs[type]);
     }
   }
 
-  // records color setting for log type, and crates convenience method ie: chronicle.info()
-  setColorForType(type, setting) {
+  // records setting and options for log type, and crates convenience method ie: chronicle.info()
+  defineType(type, setting, options=null) {
     this.color.setLogColor(type, setting);
 
+    if (options && typeof options === 'object') {
+      // sanitize user input (only store allowed types)
+      for (let option of Object.keys(options)) {
+        if (!optionKeys.includes(option)) delete options[option];
+        if (option === 'path') options[option] = this.sanitizePath(options[option]);
+      }
+
+      if (Object.keys(options).length) {
```

Comment:
I went with this approach so that we don't ever "fail". Because we need to validate user input anyway, we're already needing to iterate through all the options. Then we also need to sanitize the path if it's set, so we piggy back off of the same loop.

So it becomes a decision of whether we want to:

A. Force-crash the user's application if they provided bad options, by exiting with a throw
B. Ignore bad input, but allow the good ones. ( Old code was ignoring bad options silently, which is bad. i agree with your comment above and refactored this. )
C. Don't apply options at all if there are bad inputs, even if they're not all bad.

Personally I don't like C, and C would be the only way to do fail and bail without force crashing the app like "A".
Perhaps we should discuss the pros and cons of A & B.  I think A would be much cleaner code, but it's worth discussing the consequences of crashing a user's application.

Reply:
I realize we were already throwing in other places, so refactored this to option A